src/Psalm/Internal/Analyzer/ClassLikeAnalyzer.php

```diff
@@ -510,6 +513,37 @@ public static function getTypeFromValue($value): Type\Union
         }
     }
 
+    /**
+     * Gets the Psalm literal type from a particular value
+     *
+     * @param array|scalar|null $value
+     *
+     */
+    public static function getLiteralTypeFromValue($value): Type\Union
```

Comment:
Not sure if this function already exists somewhere and I missed it. Should this go somewhere else? Maybe as `Type::getLiteral`?